### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by stellakunzang**
Cleaned up test file structure
Added order id label to order id on profile orders show page
Deleted some things that were broken and/or not working
Add new route for Merchants items index
Login a user for tests to work
Move items methods into new controller
Remove view that doesn't serve a purpose
Add new view for creating new item that only a merchant can see
Create a new view for item index page for individual merchant
Remove useless code
Add guard statement so only merchant of this item's shop can edit and delete
Create relationship between user and merchant in model so delete merchant also deletes user
Add guard statement for adding new merchant
Add guard statements for merchant creation and edit
Fix link to go to merchant show page from admin merchant dashboard
Fix styling issues

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -1,33 +1,13 @@
...
-  def create
-    @merchant = Merchant.find(params[:merchant_id])
-    item = @merchant.items.create(item_params)
-    if item.save
-      redirect_to "/merchants/#{@merchant.id}/items"
-    else
-      flash[:error] = item.errors.full_messages.to_sentence
-      render :new
-    end
-  end
```
Comment (stellakunzang) [inline app/controllers/items_controller.rb:29]: Moved these into a new controller

File: app/controllers/merchants_items_controller.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+class MerchantsItemsController < ApplicationController
```
Comment (stellakunzang) [inline app/controllers/merchants_items_controller.rb:1]: New controller! This fixes the permissions issues we found when adding items

File: app/views/admin/merchants/show.html.erb [hunk 3]
```diff
@@ -3,7 +3,7 @@
 <p class="address"><%= @merchant.address %></p>
 <p class="address"><%= @merchant.city %>, <%= @merchant.state %> <%= @merchant.zip %></p>
 
-<h3><%= link_to "View All My Items", "/merchant/items" %><h3>
+<h3><%= link_to "View All My Items", "/merchants/#{@merchant.id}/items" %><h3>
```
Comment (stellakunzang) [inline app/views/admin/merchants/show.html.erb:6]: now when an admin wants to see a merchant's items from the admin's merchant dashboard it doesn't break things!

File: spec/features/merchants/destroy_spec.rb [hunk 4]
```diff
@@ -1,11 +1,17 @@
 require 'rails_helper'
 
-RSpec.describe "As a visitor" do
-  describe "When I visit a merchant show page" do
+RSpec.describe "As a merchant" do
```
Comment (stellakunzang) [inline spec/features/merchants/destroy_spec.rb:3]: Updated this to reflect that no anyone can destroy a merchant, only that merchant

**Review by Gallup93 — APPROVED** [review]
You rock!

File: app/controllers/items_controller.rb (hunk 1 above)
Comment (muydanny) [reply app/controllers/items_controller.rb:29]: Excellent

File: app/controllers/merchants_items_controller.rb (hunk 2 above)
Comment (muydanny) [reply app/controllers/merchants_items_controller.rb:1]: What Michael and that lady said

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -5,19 +5,6 @@ def index
...
-  def logout
-    redirect_to "/"
-  end
-
-  def new
-  end
-
-  def destroy
-  end
-
```
[the remark this answers is not in the document]
Comment (muydanny) [reply app/controllers/users_controller.rb:20]: trash can

File: app/views/admin/merchants/show.html.erb (hunk 3 above)
Comment (muydanny) [reply app/views/admin/merchants/show.html.erb:6]: And I wouldve gotten away with it too if it werent for those DEVS

**Review by muydanny — APPROVED** [review]

**Pull request merged**